File: Adda247Assignment/app/src/main/java/com/demo/myapplication/ViewModel/PostViewModel.java
package com.demo.myapplication.ViewModel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.demo.myapplication.Model.ApiResponce;

import com.demo.myapplication.Repository.ApiRepo;



public class PostViewModel extends ViewModel {
    ApiRepo apiRepo;
    MutableLiveData<ApiResponce> getPost;
    public PostViewModel()
    {

        apiRepo= new ApiRepo();
    }
    public LiveData<ApiResponce> getPostData()
    {
        if(getPost==null)
        {
            getPost= apiRepo.getPostData();
        }
        return getPost;
    }

}

File: KotlinAssignmentAdda247/app/src/main/java/com/demo/kotlinassignmentadda247/ViewModel/PostViewModel.kt
package com.demo.kotlinassignmentadda247.ViewModel

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.demo.kotlinassignmentadda247.Model.ApiResponce

import com.demo.kotlinassignmentadda247.Repository.ApiRepo
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class PostViewModel constructor(private val apiRepo: ApiRepo): ViewModel() {
    val userList= MutableLiveData<ApiResponce>()
    val errorMessage = MutableLiveData<String>()

    fun getData(){
        val responce = apiRepo.getUserData()
        responce.enqueue(object : Callback<ApiResponce>{
            override fun onResponse(call: Call<ApiResponce>, response: Response<ApiResponce>) {
                userList.postValue(response.body())
            }

            override fun onFailure(call: Call<ApiResponce>, t: Throwable) {
                errorMessage.postValue(t.message)
            }

        })
    }
}